src/schemables/WithPadding.ts

```diff
@@ -14,6 +14,18 @@ import * as Arb from '../internal/ArbitraryBase'
 import { URI as SchemaURI } from '../internal/SchemaBase'
 import * as SC from '../SchemaExt'
 import { flow, identity, pipe } from 'fp-ts/function'
+import { matchOn } from '../internal/match'
+
+/**
+ * @since 1.0.0
+ * @category Model
+ */
+export type PaddingLength =
+  | { readonly by: 'MaxLength'; readonly maxLength: number | ((s: string) => number) }
```

Comment:
this should be `MinLength` right? and flip all the comparisons for it

e.g. `MinLength = 4` leftPad n with 0s: `1 => 0001` `12314 => 12314`

Reply:
Actually, not sure about that either.  Here's [padStart](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart#parameters) from MDN:

> The length of the resulting string once the current str has been padded. If the value is less than str.length, then str is returned as-is

This makes me think that MaxLength makes more sense as a parameter, because how else will you know how much to pad?

Reply:
I guess it depends on what you're considering "max length"

that MDN quote is consistent with what I said:  
- `'1'.padStart(4, '0') === '0001'`
- `'12314'.padStart(4, '0') === '12314'`

to me, 4 is a _min_ length there since the resulting string is always _at least_ 4 characters long. it might be longer if it starts longer.

I suppose you could consider it a _max length of padding_, since you'll never pad _more than_ 4 characters: `''.padStart(4, '0') === '0000'`

imo looking at it from the length of the final string makes more sense, but I'm less concerned about that point

the comparisons are definitely wrong though. for example, `MaxLength: 4` and input is `''`, the current code

```ts
D.refine(
  (s: string): s is string => s.length <= foldUnion(maxLength)(s),
  `LeftPadding`
),
```
would say yep, `''` is valid, since its length (0) is `<= maxLength` (4)

Reply:
for what it's worth, chrome devs seem to agree with you:

![image](https://user-images.githubusercontent.com/387382/199623327-117d81a2-88ca-425d-99d0-8256e7e9cff6.png)

Reply:
That's good that I'm not crazy haha, I can really see it either way